### ISSUE TITLE
WooCommerce templates: two 404 requests when loading the templates list

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#5327]
WP makes two 404 requests when loading the Templates list (in Appearance > Editor > Templates). Those requests are:

* `/wp-json/wp/v2/themes/woocommerce`
* `/wp-json/wp/v2/plugins/WooCommerce`

While nothing seems to break because of that, we should try to avoid making unnecessary requests.

## To reproduce

1. Go to Appearance > Editor > Templates with your browser devtools Network tab open.
2. Notice there were two 404 requests:

![imatge](https://user-images.githubusercontent.com/3616980/145015265-888cc5dc-f3bf-45c5-ac3b-382068410c82.png)

## Expected behavior
We should investigate why those requests return 404 and avoid them.

**Comment by gigitux** [comment]
After some investigation, I will summarize what I discovered.

The request `/wp-json/wp/v2/plugins/WooCommerce` is done when the user lands on `Template` section, instead the request `/wp-json/wp/v2/themes/woocommerce` is done when the user lands on `Template Parts` section.

The first request [is done by this component on GB](https://github.com/WordPress/gutenberg/blob/98295006774a9bfdeaad452dd16db31fe1f837a6/packages/edit-site/src/components/list/added-by.js#L56). Since the request fails, the code uses the `slug` parameter like the fallback. [The slug value is the template theme name](https://github.com/WordPress/gutenberg/blob/98295006774a9bfdeaad452dd16db31fe1f837a6/packages/edit-site/src/components/list/added-by.js#L158). In our code we set the `template theme name` [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/f6e076ff6bb440f7baa2b1239428aa16ae6cf374/src/Utils/BlockTemplateUtils.php#L136).

More or less the same thing happens for the second request. The request is[ done by this component](https://github.com/WordPress/gutenberg/blob/98295006774a9bfdeaad452dd16db31fe1f837a6/packages/edit-site/src/components/list/added-by.js#L35). Since the request fails, the code uses the `slug` parameter like the fallback. [The slug value is the template theme name](https://github.com/WordPress/gutenberg/blob/98295006774a9bfdeaad452dd16db31fe1f837a6/packages/edit-site/src/components/list/added-by.js#L158). In this case we set the template theme name [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/f6e076ff6bb440f7baa2b1239428aa16ae6cf374/src/Utils/BlockTemplateUtils.php#L101).

In conclusion, if we want to avoid this error, we should create a REST endpoint, so GB can fetch the request correctly.

What do you think? @Aljullu

**Comment by Aljullu** [comment]
Thanks for investigating this @gigitux and for posting links to the code, that was very useful. Hope you don't mind that I continued your research. It turns out the REST endpoint already exists, it's provided by WordPress core. However it must be called in the form `/wp-json/wp/v2/plugins/woocommerce/woocommerce`. I found it out debugging the code from here:

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php

I created a fix for this in #5375. Hope you don't mind that I picked it up, but after the investigation, it made sense to directly create the PR. :pray:

**Comment by gigitux** [comment]
I'm reopening this issue because this bug will be reintroduced with #5412

First, we should be sure that `woocommerce/woocommerce` is the right endpoint.

When we are sure of this, we should check if the best option is change the theme value in our codebase (more info #5399 #5409)
